tests/dashboards/api_tests.py

```diff
@@ -146,6 +151,53 @@ def create_dashboard_with_report(self):
             db.session.delete(dashboard)
             db.session.commit()
 
+    def test_get_dashboard_charts(self):
+        """
+        Dashboard API: Test getting charts belonging to a dashboard
+        """
+        self.login(username="admin")
+        admin = self.get_user("admin")
+        slice = self.insert_chart("slice1", [admin.id], 1, params="{}")
+        dashboard = self.insert_dashboard(
+            "title", "charted", [admin.id], admin, slices=[slice]
+        )
+        uri = f"api/v1/dashboard/{dashboard.id}/charts"
+        response = self.get_assert_metric(uri, "get_charts")
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.data.decode("utf-8"))
+        self.assertEqual(data["result"], [slice.data])
+        db.session.delete(dashboard)
+        db.session.delete(slice)
+        db.session.commit()
+
+    def test_get_dashboard_charts_not_found(self):
+        """
+        Dashboard API: Test getting charts belonging to a dashboard that does not exist
+        """
+        self.login(username="admin")
+        bad_id = self.get_nonexistent_dashboard_id()
+        uri = f"api/v1/dashboard/{bad_id}/charts"
+        response = self.get_assert_metric(uri, "get_charts")
+        self.assertEqual(response.status_code, 404)
+        db.session.commit()
+
+    def test_get_dashboard_charts_empty(self):
+        """
+        Dashboard API: Test getting charts belonging to a dashboard without any charts
+        """
+        self.login(username="admin")
+        admin = self.get_user("admin")
+        dashboard = self.insert_dashboard(
+            "title", "charted", [admin.id], admin, slices=[]
+        )
+        uri = f"api/v1/dashboard/{dashboard.id}/charts"
+        response = self.get_assert_metric(uri, "get_charts")
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.data.decode("utf-8"))
```

Comment:
extract method for a reuse in all other tests

Reply:
I'm not sure what you want me to extract here

Reply:
I think that @amitmiran137 is referring, is to create a method out of:
```
        self.assertEqual(response.status_code, 200)
        data = json.loads(response.data.decode("utf-8"))
```
and reuse it, that is certainly something we should do, but on all API tests, there's lot's DRYing opportunities. If you want to take the initial shot on this one. But the entire test module should be done on a different PR, IMO

Reply:
I think there is value in keeping this explicit instead of adding a layer of indirection.